pykafka/partition.py

```diff
@@ -140,4 +142,6 @@ def update(self, brokers, metadata):
                 log.info('Updating in sync replicas list for %s', self)
                 self._isr = [brokers[b] for b in metadata.isr]
         except KeyError:
-            raise Exception("TODO: Type this exception")
+            raise LeaderNotAvailable("Leader for partition %s not available. This is "
```

Comment:
Sorry, that was sort of only half of the question I meant to put there: what I meant was, if the `KeyError` is due to a non-leader broker going missing in action, should we raise `LeaderNotAvailable` and thereby block producing to that partition, or could we swallow the exception as long as 1) there is still a leader, and maybe 2) `len(_isr) > required_acks`?

But I think that's also half off-topic for this pullreq, so we can make it a separate ticket, for if the problem comes up in practice.